subsys/bluetooth/controller/ll_sw/ll_filter.c

```diff
@@ -64,29 +142,37 @@ static void filter_clear(struct ll_filter *filter)
 	filter->addr_type_bitmask = 0;
 }
 
+
+static void filter_insert(struct ll_filter *filter, u8_t index, u8_t addr_type,
+			   u8_t *bdaddr)
+{
+	filter->enable_bitmask |= BIT(index);
+	filter->addr_type_bitmask |= ((addr_type & 0x01) << index);
+	memcpy(&filter->bdaddr[index][0], bdaddr, BDADDR_SIZE);
+}
+
+#if !defined(CONFIG_BLUETOOTH_CONTROLLER_PRIVACY)
 static u32_t filter_add(struct ll_filter *filter, u8_t addr_type, u8_t *bdaddr)
 {
-	u8_t index;
+	int index;
```

Comment:
I will change the parameter, I prefer to use ints whenever no specific width is required